test/puppetlabs/jdbc_util/core_test.clj

```diff
@@ -258,7 +258,16 @@
           (testing "but not if we call reconcile-sequence-for-column!"
             (set-sequence-value 1)
             (reconcile-sequence-for-column! test-db "sequence_test" "id")
-            (insert-dummy)))))
+            (insert-dummy))))
+
+      (testing "when the last_value of the sequence is greater than the column maximum"
+        (let [seq-max (current-sequence-value)]
+          (jdbc/execute! test-db ["DELETE FROM sequence_test"])
+          (jdbc/insert! test-db "sequence_test" {:id 1})
+          (reconcile-sequence-for-column! test-db "sequence_test" "id")
+          (testing "reconcile-sequence-for-column! doesn't change it"
+            (is (= (current-sequence-value)
```

Comment:
I don't like testing `last_value` here, because the behavior is subtly different based on `is_called`. Can we check the behavior we care about, which is that inserting a new row a) succeeds and b) gets the next id?